linux/LinuxProcessList.c

```diff
@@ -891,6 +929,13 @@ static void LinuxProcessList_readOpenVZData(LinuxProcess* process, openat_arg_t
 
 #endif
 
+static bool isContainerOrVMSlice(char *cgroup) {
+   if (String_startsWith(cgroup, "/user") || String_startsWith(cgroup, "/system") || String_startsWith(cgroup, "-"))
```

Comment:
I have made a minor change in the condition based on [this article](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/resource_management_guide/sec-default_cgroup_hierarchies).

I haven't used `if (String_startsWith(cgroup, "/machine")` since, there may be processes running in a container with a different cgroup name.

Reply:
The `cgroup` name always starts with a slash. Thus if you want to match the root slice, it should be `/-.slice` as there should be exactly one. Have a look at the raw values in the (uncompressed) `CGROUP` column for examples.

As the root slice is somewhat different, I don't think it's necessary to include it in this check. Also LXC has it's markers directly below the root -> `/lxc.payload.somecontainername/system.slice/console-getty.service`.